packages/next-intl/test/middleware/middleware.test.tsx

```diff
@@ -531,6 +537,26 @@ describe('prefix-based routing', () => {
         );
       });
 
+      it('redirects an invalid, upper-cased request for a localized route to the case-sensitive format', () => {
+        middlewareWithPathnames(createMockRequest('/DE-AT', 'de-AT'));
+        expect(MockedNextResponse.rewrite).toHaveBeenCalled();
+        expect(MockedNextResponse.next).not.toHaveBeenCalled();
+        expect(MockedNextResponse.redirect).not.toHaveBeenCalled();
+        expect(MockedNextResponse.rewrite.mock.calls[0][0].toString()).toBe(
+          'http://localhost:3000/de-AT'
```

Comment:
I'm not sure if we're on the same page, but this test doesn't seem to assert for what the description says.

Let's clear up the terminology:
- **Redirect**: A status code in the 3xx range is returned that points the user to another location.
- **Rewrite**: The incoming request is mapped to another internal path.

In the context of `next-intl`, we e.g. have a _redirect_ for the root `/` to the default locale (e.g. `/en`). On the other hand, we use _rewrites_ e.g. for [localized pathnames](https://next-intl-docs.vercel.app/docs/routing/middleware#localizing-pathnames) to handle them via a single internal pathname.

The way I'd imagine this feature and how [I've observed it e.g. on MDN](https://github.com/amannn/next-intl/issues/775#issuecomment-1930415643) is that a redirect is returned when there's a case mismatch.

Due to this, the test should read like this:

```suggestion
        expect(MockedNextResponse.rewrite).not.toHaveBeenCalled();
        expect(MockedNextResponse.next).not.toHaveBeenCalled();
        expect(MockedNextResponse.redirect).toHaveBeenCalled();
        expect(MockedNextResponse.redirect.mock.calls[0][0].toString()).toBe(
          'http://localhost:3000/de-AT'
```

Reply:
Ah yes, that makes perfect sense. I also noticed during my research that a redirect is returned if there is a mismatch in the locale - I got something mixed up when writing the tests 😄. I have now adjusted this according to your suggestions and the tests failed as expected, but after I undid the changes in `resolveLocale`, everything works. And as a result, only very few changes are necessary in the middleware 🙏🏼.